cpp/oneapi/dal/algo/kmeans/test/badarg.cpp

```diff
@@ -0,0 +1,159 @@
+    }
+
+private:
+    static constexpr std::array<float, element_count> train_data_ = {
+        1.0, 1.0, 2.0, 2.0, 1.0, 2.0, 2.0, 1.0, -1.0, -1.0, -1.0, -2.0, -2.0, -1.0, -2.0, -2.0
+    };
+
+    static constexpr std::array<float, bad_element_count> infer_data_ = {
+        1.0, 1.0, 2.0, 2.0, 1.0, 2.0, 2.0, 1.0, -1.0, -1.0, -1.0, -2.0, -2.0, -1.0, -2.0, -2.0
+    };
+
+    static constexpr std::array<float, bad_element_count> bad_initial_centroids = { 0.0, 0.0, 0.0,
+                                                                                    0.0, 0.0, 0.0 };
+
+    static constexpr std::array<float, bad_element_count> initial_centroids = { 0.0,
+                                                                                0.0,
+                                                                                0.0,
+                                                                                0.0 };
+};
+
+#define KMEANS_BADARG_TEST(name) \
+    TEMPLATE_TEST_M(kmeans_badarg_test, name, "[kmeans][badarg]", method::lloyd_dense)
+
+KMEANS_BADARG_TEST("accepts positive cluster_count") {
+    REQUIRE_NOTHROW(this->get_descriptor().set_cluster_count(1));
+}
+
+KMEANS_BADARG_TEST("throws if cluster_count is negative") {
+    REQUIRE_THROWS_AS(this->get_descriptor().set_cluster_count(-1), domain_error);
+}
+
+KMEANS_BADARG_TEST("throws if cluster_count is zero") {
+    REQUIRE_THROWS_AS(this->get_descriptor().set_cluster_count(0), domain_error);
+}
+
+KMEANS_BADARG_TEST("accepts positive max iteration count") {
+    REQUIRE_NOTHROW(this->get_descriptor().set_max_iteration_count(1));
+}
+
+KMEANS_BADARG_TEST("accepts zero max iteration count") {
+    REQUIRE_NOTHROW(this->get_descriptor().set_max_iteration_count(0));
+}
+
+KMEANS_BADARG_TEST("throws if max iteration count is negative") {
+    REQUIRE_THROWS_AS(this->get_descriptor().set_max_iteration_count(-1), domain_error);
+}
+
+KMEANS_BADARG_TEST("accepts positive accuracy threshold") {
+    REQUIRE_NOTHROW(this->get_descriptor().set_accuracy_threshold(0.01));
+}
+
+KMEANS_BADARG_TEST("accepts zero accuracy threshold") {
+    REQUIRE_NOTHROW(this->get_descriptor().set_accuracy_threshold(0.0));
+}
+
+KMEANS_BADARG_TEST("throws if accuracy threshold is negative") {
+    REQUIRE_THROWS_AS(this->get_descriptor().set_accuracy_threshold(-0.1), domain_error);
+}
+
+KMEANS_BADARG_TEST("throws if train data is empty") {
+    const auto kmeans_desc = this->get_descriptor().set_cluster_count(2);
+
+    REQUIRE_THROWS_AS(this->train(kmeans_desc, homogen_table{}), domain_error);
+}
+
+KMEANS_BADARG_TEST("throws if initial centroids rows less than cluster count") {
+    const auto kmeans_desc = this->get_descriptor().set_cluster_count(this->huge_cluster_count);
+
+    REQUIRE_THROWS_AS(train(kmeans_desc, this->get_train_data(), this->get_initial_centroids()),
+                      invalid_argument);
+}
+
+KMEANS_BADARG_TEST("throws if train data columns neq initial centroid columns") {
```

Comment:
"neq"?

Reply:
It should be equal so neq is a condition to throw, right?